### PR TITLE
feat: implement 0x15 - ISZERO opcode

### PR DESCRIPTION
**Pull request by Noeljarillo**
Implemented the OP code iszero

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ X] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### CODE REVIEW

File: crates/evm/src/tests/test_instructions/test_comparison_operations.cairo [hunk 1]
```diff
@@ -165,6 +165,21 @@ fn test_not_random_uint() {
     );
 }
 
+#[test]
+#[available_gas(20000000)]
+fn test_is_zero() {
```
Comment (enitrat) [inline crates/evm/src/tests/test_instructions/test_comparison_operations.cairo:170]:
```suggestion
fn test_is_zero_true() {
```

Can you also add another test 


`test_is_zero_false`

File: crates/evm/src/instructions/comparison_operations.cairo [hunk 2]
```diff
@@ -54,7 +54,13 @@ impl ComparisonAndBitwiseOperations of ComparisonAndBitwiseOperationsTrait {
     /// 0x15 - ISZERO
     /// # Specification: https://www.evm.codes/#15?fork=shanghai
     fn exec_iszero(ref self: ExecutionContext) -> Result<(), EVMError> {
-        Result::Ok(())
+        let popped = self.stack.pop_n(1)?;
```
Comment (enitrat) [inline crates/evm/src/instructions/comparison_operations.cairo:57]:
```suggestion
        let popped = self.stack.pop()?;
```

File: crates/evm/src/tests/test_instructions/test_comparison_operations.cairo (hunk 1 above)
Comment (Noeljarillo) [reply crates/evm/src/tests/test_instructions/test_comparison_operations.cairo:170]: got it working on it now!

**Review by Noeljarillo — COMMENTED** [review]
added suggested change, and false test case

File: crates/evm/src/tests/test_instructions/test_comparison_operations.cairo [hunk 3]
```diff
@@ -165,6 +165,36 @@ fn test_not_random_uint() {
...
+    // Given
+    let mut ctx = setup_execution_context();
+    ctx.stack.push(0x00).unwrap();
+    
+    // When
+    ctx.exec_iszero();
+
+    // Then
+    assert(ctx.stack.len() == 1, 'stack should have one element');
+    assert(ctx.stack.peek().unwrap() == 0x01, 'stack top should be 0x01');
```
Comment (ClementWalter) [inline crates/evm/src/tests/test_instructions/test_comparison_operations.cairo:180]: can you put a more meaningful error message, like "stack top should be true" instead?

File: crates/evm/src/tests/test_instructions/test_comparison_operations.cairo [hunk 4]
```diff
@@ -165,6 +165,36 @@ fn test_not_random_uint() {
...
+    // Given
+    let mut ctx = setup_execution_context();
+    ctx.stack.push(0x01).unwrap();
+    
+    // When
+    ctx.exec_iszero();
+
+    // Then
+    assert(ctx.stack.len() == 1, 'stack should have one element');
+    assert(ctx.stack.peek().unwrap() == 0x00, 'stack top should be 0x00');
```
Comment (ClementWalter) [inline crates/evm/src/tests/test_instructions/test_comparison_operations.cairo:195]: same here, with "false" plz

**Review by ClementWalter — CHANGES_REQUESTED** [review]
I think that you need to format as well

**Review by enitrat — APPROVED** [review]
LGtM!

**Review by enitrat — APPROVED** [review]

**Review by ClementWalter — APPROVED** [review]

**Pull request merged**